app/javascript/components/clients/Doctorappointment.js

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable no-restricted-globals */
+/* eslint-disable no-alert */
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+
+import './Bookappointment.css';
+
+const DoctorAppointment = () => {
+  const { id } = useParams();
+  const appointments = useSelector((state) => state.appointmentReducer);
+
+  const onSubmit = (e) => {
+    document.querySelector('.popup-box').style.display = 'none';
+    const date = document.getElementById('appointment_date').value;
+    const time = document.getElementById('appointment_time').value;
+    const doctorId = parseInt(id, 10);
+    if ((appointments.filter((appointment) => appointment.appointment_time === time).length > 0)
+    && (appointments.filter((appointment) => appointment.appointment_date === date).length > 0)
+    && (appointments.filter((appointment) => appointment.doctor_id === doctorId).length > 0)) {
+      alert('appointment not available');
+      e.preventDefault();
+    }
+  }
+
+  return (
+    <div className="popup-box">
+      <form action="http://localhost:3000/api/v1/appointments" method="post" className="box form-group">
+        <div>
+          <h4>Choose a date and time</h4>
+        </div>
+        <div className="close-btn d-flex">
+          <span className="close-icon" onClick={onSubmit}>x</span>
+        </div>
+        {DoctorAppointment.content}
+        <input className="date-input" type="date" id="appointment_date" name="exampleInputDate1" />
```

Comment:
Our form elements should be "controlled components". Their values should be local states.